tasks/docker-engine.yml

```diff
@@ -19,19 +19,60 @@
     repo: 'deb https://apt.dockerproject.org/repo ubuntu-{{ ansible_distribution_release }} main'
     update_cache: yes
 
+- name: see if docker is already installed
+  stat: path=/usr/bin/docker get_checksum=False
+  register: docker_exec_stat_result
+  always_run: True 
 
-# While wildcarding the version is cute, in practice, there only seems to be one
-# version of Docker available at a time. I'm not sure what will happen on the next
-# release. Alternatively, we could install LXC Docker, which does have many more
-# versions, but not the most recent versions.
+- name: Determine installed version
```

Comment:
Since we have a specific use case for using this role to upgrade from 1.5.0, I figured always using "docker version" was better than trying "docker --version" and falling back to "docker version".